query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java

```diff
@@ -35,7 +35,7 @@ CompletionStage<QueryResponse> perform(BitSet segments) {
    public LuceneSearchResult<Object> fetchIds(SearchQueryBuilder query) {
       long start = queryStatistics.isEnabled() ? System.nanoTime() : 0;
 
-      LuceneSearchResult<Object> result = query.ids().fetch(queryDefinition.getMaxResults());
+      LuceneSearchResult<Object> result = query.ids().fetchAll();
```

Comment:
You used ' fetchAllHits ' in the main branch. Shouldn't the same method be used here, too?

Reply:
The change you mentioned here already present in this other contribution:  https://github.com/infinispan/infinispan/pull/13156.
As I mentioned above: both are required!